bigquery/google/cloud/bigquery/client.py

```diff
@@ -1929,6 +1935,56 @@ def list_rows(
         )
         return row_iterator
 
+    def _schema_from_json_file_object(self, file_obj):
+        """Helper function for schema_from_json that takes a
+       file object that describes a table schema.
+
+       Returns:
+            List of schema field objects.
+        """
+        json_data = json.load(file_obj)
+        return [SchemaField.from_api_repr(field) for field in json_data]
+
+    def _schema_to_json_file_object(self, schema_list, file_obj):
+        """Helper function for schema_to_json that takes a schema list and file
+        object and writes the schema list to the file object with json.dump
+        """
+        json.dump(schema_list, file_obj, indent=2, sort_keys=True)
+
+    def schema_from_json(self, file_or_path):
+        """Takes a file object or file path that contains json that describes
+        a table schema.
+
+        Returns:
+            List of schema field objects.
+        """
+        if isinstance(file_or_path, io.IOBase):
+            return self._schema_from_json_file_object(file_or_path)
+
+        try:
+            with open(file_or_path) as file_obj:
+                return self._schema_from_json_file_object(file_obj)
+        except OSError:
```

Comment:
Oops, coverage is failing on this line and the similar line in the other function. We'd need a test where `open()` fails.

Honestly, I'd be okay removing the `try` block and letting these errors just raise, too.

Reply:
Removed the `try` block as suggested for both functions.